### PR TITLE
Lookup by date now working

### PR DESCRIPTION
**Pull request by Anelon**
Both in the rest and in the site (which uses the rest controller) takes date format specified by the planning team

### CODE REVIEW

File: src/main/java/cst438flights/service/FlightService.java [hunk 1]
```diff
@@ -26,6 +30,10 @@
     @Autowired
     private FanoutExchange fanout;
 
+    //date formatter for conversion of date types
+    static private SimpleDateFormat travelDate = new SimpleDateFormat("MM/dd/yyyy");
+    static private SimpleDateFormat sqlDate = new SimpleDateFormat("yyyy-MM-dd");
+
```
Comment (Nhinds88) [inline src/main/java/cst438flights/service/FlightService.java:36]: Good work with the date formatting

File: src/main/java/cst438flights/service/FlightService.java [hunk 2]
```diff
@@ -48,8 +56,31 @@
...
+                //save new date as a string
+                Date endDate = cal.getTime();
+
+                return flightRepository.findByDepartureArrivalAirport(departureAirport, arrivalAirport, startDate, endDate);
+            } catch (Exception e) {
+                System.out.println(e);
+                System.out.println("An error occured");
+                return null;
+            }
+        }
```
Comment (Nhinds88) [inline src/main/java/cst438flights/service/FlightService.java:83]: I like this exception call, it'll be useful for the rest controller

**Review by Nhinds88 — APPROVED** [review]
Looks good, great job with the date selection!

File: src/main/resources/templates/index.html [hunk 3]
```diff
@@ -165,14 +165,14 @@ <h3>Look up previous flights</h3>
     }
     function dateUpdate() {
         lookupDate = document.getElementById("lookupDate").value;
-        console.log(lookupDate);
         let dateObj = new Date(lookupDate);
+        //get the timezone offset
```
Comment (espresso64) [inline src/main/resources/templates/index.html:169]: nice work with the offset, fixed the calendar issue perfectly

File: src/main/java/cst438flights/controller/flightRestController.java [hunk 4]
```diff
@@ -46,13 +46,40 @@
...
+        List<Flight> flightInfo = flightService.getAvailableFights(departureAirport, arrivalAirport, date);
+        System.out.println(flightInfo);
+
+        if (flightInfo == null) {
+            return new ResponseEntity<List<Flight>>(HttpStatus.NOT_FOUND);
+        } else {
+            return new ResponseEntity<List<Flight>>(flightInfo, HttpStatus.OK);
+        }
+    }
+
```
Comment (espresso64) [inline src/main/java/cst438flights/controller/flightRestController.java:68]: Added the extra API option that allows the planner to enter a date. This addition to the code is standalone and should not affect any existing code. Andrew, thanks for the idea!

**Review by espresso64 — APPROVED** [review]
Code looks great! Excellent work on the date system implementation.

**Pull request merged**